src/util/kaldi-table-inl.h

```diff
@@ -302,6 +325,24 @@ template<class Holder>  class SequentialTableReaderScriptImpl:
       if (!key_.empty() && !data_rxfilename_.empty()) {
         // Got a valid line.
         state_ = kHaveScpLine;
+        // Here we "peel off" the range_ specifer (if any, e.g. [1:2,2:10])
+        // from data_rxfilename_ (e.g. 1.ark:100).
+        if (data_rxfilename_.find_last_of("]") == data_rxfilename_.length() - 1) {
+          std::vector<std::string> splits;
+          SplitStringToVector(data_rxfilename_, "[]", true, &splits);
```

Comment:
Better to spend a little more effort to parse it exactly- this will accept too many wrong things.  If the code gets too long you can declare a function for this.